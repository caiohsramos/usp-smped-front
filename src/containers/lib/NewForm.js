import React from 'react';
import { Container } from './Container';
import { NewFormController } from '../../controllers';
import { FormItem } from '../../components';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from '@material-ui/core/FormControl';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';
import Button from '@material-ui/core/Button';

class NewForm extends Container {
	
	constructor(props) {
		super(props);

		this.state = {
			VERSION: 1,
			OWNER: '',
			NAME: '',
			OFFICE: 'smped',
			ACTIVITY: '',
			FIELDS: [],	
		};

		const offices = [
			{
				value: 'smped',
				label: 'SMPED'
			},
			{
				value: 'smdhc',
				label: 'SMDHC'
			}
		];

		const toController = {
			callback: this.callback,
			getState: this.getState,
			getProps: this.getProps,
			router: props.router
		};

		this.controller = new NewFormController(toController);
		this.offices = offices;
	}

	componentDidUpdate(){
		console.log(this.state);
	}

	render() {

		return (
			<div className="container newFormContainer">
				<Grid container spacing={24}>
					<Grid item xs={12}>
						<Paper>
							<h1>Novo formulário</h1>
							<FormControl className="field-name">
								<InputLabel htmlFor="name-simple">Nome do formulário</InputLabel>
								<Input 
								id="form-name" 
								value={this.state.NAME} 
								onChange={this.controller.handleChange('NAME')} />
							</FormControl>
							<TextField
								id="office"
								select
								label="Secretaria"
								className="select-type"
								value={this.state.OFFICE}
								onChange={this.controller.handleChange('OFFICE')}
								margin="normal">
								{this.offices.map(option => (
									<MenuItem key={option.value} value={option.value}>
										{option.label}
									</MenuItem>
								))}
							</TextField>
							<FormControl className="field-name">
								<InputLabel htmlFor="name-simple">Atividade</InputLabel>
								<Input
									id="form-activity"
									value={this.state.ACTIVITY}
									onChange={this.controller.handleChange('ACTIVITY')} />
							</FormControl>
							
							<div aria-live="polite">
								{
									this.state.FIELDS.map(formItem => (
										<FormItem
											key={formItem.ORDER}
											formItem={formItem}
											handleChangeFormItem={this.controller.handleChangeFormItem}
											/> 
										
									))

								
							}
							</div>
							<div className="row">
								<Button
									id='novo-item'
									variant='contained'
									color='primary'
									onClick={this.controller.addFormItem}>
									Novo item
								</Button>
								<Button 
									id='remover-item'
									color='secondary'
									variant='contained'
									onClick={this.controller.removeFormItem}>
									Remover Item
								</Button>
							</div>
						</Paper>
					</Grid>
				</Grid>
			</div>
		);
	}


}

export default (NewForm);
