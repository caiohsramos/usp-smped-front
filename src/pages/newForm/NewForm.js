import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Container } from '../Container';
import { NewFormController } from './NewFormController';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';
import { FormItem } from './components/FormItem';
import Button from '@material-ui/core/Button';
import SaveIcon from '@material-ui/icons/Save';
import Snackbar from '@material-ui/core/Snackbar';
import { setMsg } from '../../actions/MessageActions';

class NewForm extends Container {

    constructor(props) {
        super(props);

        this.state = {
            version: '1',
            owner: '',
            name: '',
            office: 'smped',
            activity: '',
            fields: [],
            errors: {},
            snack: { open: false, msg: '', success: false }
        };

        const offices = [
            {
                value: 'smped',
                label: 'SMPED'
            },
            {
                value: 'smdhc',
                label: 'SMDHC'
            }
        ];

        const toController = {
            callback: this.callback,
            getState: this.getState,
            getProps: this.getProps,
            router: props.router,
            setMsg: props.setMsg
        };

        this.controller = new NewFormController(toController);
        this.offices = offices;
    }

    componentDidUpdate() {
    }

    render() {
        const { snackHandleClose } = this.controller;
        return (
            <div className="newFormContainer">
                <Snackbar
                    anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
                    open={(this.state.snack.open)}
                    onClose={snackHandleClose}
                    className={this.state.snack.success ? 'snack-success' : 'snack-fail'}
                    ContentProps={{
                        'aria-describedby': 'message-id',
                    }}
                    message={<span id="message-id">{this.state.snack.msg}</span>}
                />
                <Paper>
                    <Grid container spacing={16} alignItems="center" justify='space-evenly'>
                        <Grid item container justify='center'>
                            <Grid><h1>Novo formulário</h1></Grid>
                        </Grid>
                        <Grid item zeroMinWidth>
                            <TextField
                                id="form-name"
                                label="Nome do formulário"
                                className="field-name"
                                value={this.state.name}
                                onChange={this.controller.handleChange('name')}
                                margin="normal"
                                error={this.state.errors["form-name"]}
                            />
                        </Grid>
                        <Grid item>
                            <TextField
                                id="office"
                                select
                                label="Secretaria"
                                className="select-type"
                                value={this.state.office}
                                onChange={this.controller.handleChange('office')}
                                margin="normal">
                                {this.offices.map(option => (
                                    <MenuItem key={option.value} value={option.value}>
                                        {option.label}
                                    </MenuItem>
                                ))}
                            </TextField>
                        </Grid>
                        <Grid item zeroMinWidth>
                            <TextField
                                id="form-activity"
                                label="Atividade"
                                className="field-name"
                                value={this.state.activity}
                                onChange={this.controller.handleChange('activity')}
                                margin="normal"
                                error={this.state.errors["form-activity"]}
                            />
                        </Grid>
                        <Grid item container direction='column' >
                            <div aria-live="polite">
                                {
                                    this.state.fields.map(formItem => (
                                        <Grid item key={formItem.order} >
                                            <FormItem
                                                key={formItem.order}
                                                formItem={formItem}
                                                handleChangeFormItem={this.controller.handleChangeFormItem}
                                                errors={this.state.errors}
                                            />
                                        </Grid>
                                    ))
                                }
                            </div>
                        </Grid>
                    </Grid>
                    <Grid container justify='space-evenly' alignItems='center'>
                        <Button
                            id='novo-btn'
                            variant='contained'
                            color='primary'
                            onClick={this.controller.addFormItem}>
                            Novo item
                        </Button>
                        <Button
                            id='remover-btn'
                            color='secondary'
                            variant='contained'
                            disabled={!(this.state.fields.length)}
                            onClick={this.controller.removeFormItem}>
                            Remover Item
                        </Button>
                    </Grid>
                    <Grid container id='fields-grid' justify='center'>
                        <Button
                            id='salvar-btn'
                            color='primary'
                            variant='contained'
                            disabled={!(this.state.fields.length)}
                            onClick={this.controller.submitForm}>
                            Salvar formulário
                            <SaveIcon />
                        </Button>
                    </Grid>
                </Paper>
            </div>
        );
    }


}

const mstp = (state) => {
    return {
        token: state.session.accessToken,
    };
};

const mdtp = dispatch =>
    bindActionCreators({ setMsg }, dispatch);

export default connect(mstp, mdtp)(NewForm);
