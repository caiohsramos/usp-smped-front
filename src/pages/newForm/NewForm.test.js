import React from 'react';
import NewForm from './NewForm';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import reducer from '../../reducers';
import { mount, shallow, simulate } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const store = createStore(reducer);
Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer
        .create(<NewForm store={store} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render one form field when click new field button', () => {
    // To understand the next line, see this:
    // https://github.com/airbnb/enzyme/issues/431
    const wrapper = mount(shallow(<NewForm store={store} />).get(0));
    let createItemBtn = wrapper.find('#novo-btn').hostNodes();
    createItemBtn.simulate('click');
    createItemBtn.simulate('click');
    let fieldsSize = wrapper.state().fields.length
    expect(fieldsSize).toBe(2);
})
