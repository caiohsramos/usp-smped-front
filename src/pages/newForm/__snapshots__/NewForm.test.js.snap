// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="newFormContainer"
>
  <div
    className="MuiPaper-root-8 MuiPaper-elevation2-12 MuiPaper-rounded-9"
  >
    <div
      className="MuiGrid-container-35 MuiGrid-spacing-xs-16-58 MuiGrid-align-items-xs-center-43 MuiGrid-justify-xs-space-evenly-56"
    >
      <div
        className="MuiGrid-container-35 MuiGrid-item-36 MuiGrid-justify-xs-center-52"
      >
        <div
          className=""
        >
          <h1>
            Novo formulário
          </h1>
        </div>
      </div>
      <div
        className="MuiGrid-item-36 MuiGrid-zeroMinWidth-37"
      >
        <div
          className="MuiFormControl-root-132 MuiFormControl-marginNormal-133 field-name"
        >
          <label
            className="MuiFormLabel-root-147 MuiInputLabel-root-136 MuiInputLabel-formControl-141 MuiInputLabel-animated-144"
            data-shrink={false}
            htmlFor="form-name"
          >
            Nome do formulário
          </label>
          <div
            className="MuiInputBase-root-167 MuiInput-root-154 MuiInput-underline-158 MuiInputBase-formControl-168 MuiInput-formControl-155"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-177 MuiInput-input-162"
              disabled={false}
              id="form-name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-36"
      >
        <div
          className="MuiFormControl-root-132 MuiFormControl-marginNormal-133 select-type"
        >
          <label
            className="MuiFormLabel-root-147 MuiFormLabel-filled-151 MuiInputLabel-root-136 MuiInputLabel-formControl-141 MuiInputLabel-animated-144 MuiInputLabel-shrink-143"
            data-shrink={true}
            htmlFor="office"
          >
            Secretaria
          </label>
          <div
            className="MuiInputBase-root-167 MuiInput-root-154 MuiInput-underline-158 MuiInputBase-formControl-168 MuiInput-formControl-155"
            onClick={[Function]}
          >
            <div
              className="MuiSelect-root-184"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-185 MuiSelect-selectMenu-188 MuiInputBase-input-177 MuiInput-input-162"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                SMPED
              </div>
              <input
                id="office"
                type="hidden"
                value="smped"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-191 MuiSelect-icon-190"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-36 MuiGrid-zeroMinWidth-37"
      >
        <div
          className="MuiFormControl-root-132 MuiFormControl-marginNormal-133 field-name"
        >
          <label
            className="MuiFormLabel-root-147 MuiInputLabel-root-136 MuiInputLabel-formControl-141 MuiInputLabel-animated-144"
            data-shrink={false}
            htmlFor="form-activity"
          >
            Atividade
          </label>
          <div
            className="MuiInputBase-root-167 MuiInput-root-154 MuiInput-underline-158 MuiInputBase-formControl-168 MuiInput-formControl-155"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-177 MuiInput-input-162"
              disabled={false}
              id="form-activity"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-container-35 MuiGrid-item-36 MuiGrid-direction-xs-column-38"
      >
        <div
          aria-live="polite"
        />
      </div>
    </div>
    <div
      className="MuiGrid-container-35 MuiGrid-align-items-xs-center-43 MuiGrid-justify-xs-space-evenly-56"
    >
      <button
        className="MuiButtonBase-root-230 MuiButton-root-204 MuiButton-contained-215 MuiButton-containedPrimary-216 MuiButton-raised-218 MuiButton-raisedPrimary-219"
        disabled={false}
        id="novo-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-205"
        >
          Novo item
        </span>
        <span
          className="MuiTouchRipple-root-233"
        />
      </button>
      <button
        className="MuiButtonBase-root-230 MuiButtonBase-disabled-231 MuiButton-root-204 MuiButton-contained-215 MuiButton-containedSecondary-217 MuiButton-raised-218 MuiButton-raisedSecondary-220 MuiButton-disabled-224"
        disabled={true}
        id="remover-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-205"
        >
          Remover Item
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-container-35 MuiGrid-justify-xs-center-52"
      id="fields-grid"
    >
      <button
        className="MuiButtonBase-root-230 MuiButtonBase-disabled-231 MuiButton-root-204 MuiButton-contained-215 MuiButton-containedPrimary-216 MuiButton-raised-218 MuiButton-raisedPrimary-219 MuiButton-disabled-224"
        disabled={true}
        id="salvar-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-205"
        >
          Salvar formulário
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-191"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
